rustls/src/client/tls13.rs

```diff
@@ -88,14 +88,47 @@ pub(super) fn handle_server_hello(
             )
         })?;
 
-    if our_key_share.group() != their_key_share.group {
+    // Determine if the primary (`our_key_share`) or secondary
+    // (`our_key_share::hybrid_pair()`) was selected.
+
+    let our_key_share = if our_key_share.group() == their_key_share.group {
+        // primary
+        our_key_share
+    } else if our_key_share
+        .hybrid_pair()
+        .map(|pair| pair.group())
+        == Some(their_key_share.group)
+    {
+        let secondary = our_key_share
+            .into_hybrid_pair()
+            .expect("into_hybrid_pair() must by supported if hybrid_pair() is");
```

Comment:
Thinking back to my not fully baked review comment, I think this `expect()` might fall away if the `hybrid_pair()` return was also sufficiently capable of being turned into the `Box`'d form we need.